include/convex_bodies/spectrahedron.h

```diff
@@ -48,8 +54,24 @@ class LMI {
 
        for (Iter iter=matrices.begin() ; iter!=matrices.end() ; iter++, i++)
            res += x(i) * (*iter);
+
+       return res;
     }
 
+    bool isNegativeDefinite(VT& x) {
+        MT mt = evaluate(x);
+
+        Eigen::EigenSolver<MT> solver;
+        solver.compute(mt);
+        Eigen::GeneralizedEigenSolver<MT>::ComplexVectorType eivals = solver.eigenvalues();
```

Comment:
Why do we need ComplexVectorType here? If the matrices are symmetric, then eigenvalues are all real.

Reply:
The Eigen library requires it. In the generalized eigenvalues as well, I must use ComplexVectorType.